pkg/backend/token_authcode.go

```diff
@@ -41,29 +38,41 @@ type refreshDescriptor struct {
 var _ scheduler.Descriptor = &refreshDescriptor{}
 
 func (rd *refreshDescriptor) Run(ctx context.Context, pc chan<- scheduler.Process) error {
-	ticker := rd.backend.clock.NewTicker(refreshInterval)
-	defer ticker.Stop()
+	// start out checking once per minute
+	refreshInterval := 60
+	ticker := rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
 
 	for {
-		err := rd.backend.data.Managers(rd.storage).AuthCode().ForEachAuthCodeKey(ctx, func(keyer persistence.AuthCodeKeyer) {
-			proc := &refreshProcess{
-				backend: rd.backend,
-				storage: rd.storage,
-				keyer:   keyer,
+		c, err := rd.backend.getCache(ctx, rd.storage)
+		if err == nil && c != nil && c.Config.RefreshInterval != 0 {
+			if c.Config.RefreshInterval != refreshInterval {
+				refreshInterval = c.Config.RefreshInterval
+				ticker.Stop()
+				ticker = rd.backend.clock.NewTicker(time.Duration(refreshInterval) * time.Second)
 			}
-
-			select {
-			case pc <- proc:
-			case <-ctx.Done():
+			err = rd.backend.data.Managers(rd.storage).AuthCode().ForEachAuthCodeKey(ctx, func(keyer persistence.AuthCodeKeyer) {
+				proc := &refreshProcess{
+					backend: rd.backend,
+					storage: rd.storage,
+					keyer:   keyer,
+					seconds: refreshInterval + 10,
+				}
+
+				select {
+				case pc <- proc:
+				case <-ctx.Done():
+				}
+			})
+			if err != nil {
+				ticker.Stop()
```

Comment:
You can change the defer you removed above from `defer ticker.Stop()` to `defer func() { ticker.Stop() }` if you'd like to have it use a reference to whatever the current value of `ticker` is. That would avoid having to remember to put this in every case where this method returns. Example: https://play.golang.org/p/gJJ7_tjV140 (remember defers are a stack so run LIFO).

Reply:
Very good.